pkgs/development/python-modules/vispy/default.nix

```diff
@@ -31,6 +31,13 @@ buildPythonPackage rec {
     hash = "sha256-77u4R6kIuvfnFpq5vylhOKOTZPNn5ssKjsA61xaZ0x0=";
   };
 
+  postPatch = ''
+    # https://numpy.org/devdocs/dev/depending_on_numpy.html#numpy-2-0-specific-advice
+    # upstream enforce builds with numpy 2+, which is backward compat with 1.xx
+    substituteInPlace pyproject.toml \
+      --replace-warn "numpy>=2.0.0rc2" "numpy"
```

Comment:
any reason not to use `--replace-fail` ?

Reply:
gut feeling, changed it to `fail`